# Covid-19 Impact Analysis (Case Study)

The outbreak of the deadly virus i.e, Covid-19 impacted many lives all around the globe. Almost all the countries were impacted 
negatively by this outbreak. Here we will get to know the adverse impact on Covid-19.

The dataset that we are using contains about:
1. Country code
2. Name of all the countries
3. Date of the record
4. Human development index of all the countries
5. Daily covid-19 cases
6. Daily deaths due to covid-19
7. Stringency index of the countries
8. Population of the country
9. GDP per capita of the country

In [219]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

In [220]:
data = pd.read_csv('transformed_data.csv')
data2 = pd.read_csv('raw_data.csv')

In [221]:
data.head()

CODE      COUNTRY        DATE    HDI   TC   TD  STI        POP    GDPCAP
0  AFG  Afghanistan  2019-12-31  0.498  0.0  0.0  0.0  17.477233  7.497754
1  AFG  Afghanistan  2020-01-01  0.498  0.0  0.0  0.0  17.477233  7.497754
2  AFG  Afghanistan  2020-01-02  0.498  0.0  0.0  0.0  17.477233  7.497754
3  AFG  Afghanistan  2020-01-03  0.498  0.0  0.0  0.0  17.477233  7.497754
4  AFG  Afghanistan  2020-01-04  0.498  0.0  0.0  0.0  17.477233  7.497754

In [222]:
data2.head()

iso_code     location        date  total_cases  total_deaths  \
0      AFG  Afghanistan  2019-12-31          0.0           0.0   
1      AFG  Afghanistan  2020-01-01          0.0           0.0   
2      AFG  Afghanistan  2020-01-02          0.0           0.0   
3      AFG  Afghanistan  2020-01-03          0.0           0.0   
4      AFG  Afghanistan  2020-01-04          0.0           0.0   

   stringency_index  population  gdp_per_capita  human_development_index  \
0               0.0    38928341        1803.987                    0.498   
1               0.0    38928341        1803.987                    0.498   
2               0.0    38928341        1803.987                    0.498   
3               0.0    38928341        1803.987                    0.498   
4               0.0    38928341        1803.987                    0.498   

  Unnamed: 9 Unnamed: 10 Unnamed: 11  Unnamed: 12  Unnamed: 13  
0      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
1      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
2      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
3      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
4      #NUM!       #NUM!       #NUM!    17.477233  7.497754494

In [223]:
data.shape

(50418, 9)

In [224]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50418 entries, 0 to 50417
Data columns (total 9 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   CODE     50418 non-null  object 
 1   COUNTRY  50418 non-null  object 
 2   DATE     50418 non-null  object 
 3   HDI      44216 non-null  float64
 4   TC       50418 non-null  float64
 5   TD       50418 non-null  float64
 6   STI      50418 non-null  float64
 7   POP      50418 non-null  float64
 8   GDPCAP   50418 non-null  float64
dtypes: float64(6), object(3)
memory usage: 3.5+ MB


# Data Cleaning

Data Cleaning is the primary stage of Data analysis.This stage involves removal of unwanted columns, identifying the missing values and changing the datatype of the column. 

In [225]:
data["HDI"].fillna(method = 'ffill', inplace = True)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50418 entries, 0 to 50417
Data columns (total 9 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   CODE     50418 non-null  object 
 1   COUNTRY  50418 non-null  object 
 2   DATE     50418 non-null  object 
 3   HDI      50418 non-null  float64
 4   TC       50418 non-null  float64
 5   TD       50418 non-null  float64
 6   STI      50418 non-null  float64
 7   POP      50418 non-null  float64
 8   GDPCAP   50418 non-null  float64
dtypes: float64(6), object(3)
memory usage: 3.5+ MB


In [226]:
data['TC'] = data['TC'].astype(int)
data['TD'] = data['TD'].astype(int)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50418 entries, 0 to 50417
Data columns (total 9 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   CODE     50418 non-null  object 
 1   COUNTRY  50418 non-null  object 
 2   DATE     50418 non-null  object 
 3   HDI      50418 non-null  float64
 4   TC       50418 non-null  int32  
 5   TD       50418 non-null  int32  
 6   STI      50418 non-null  float64
 7   POP      50418 non-null  float64
 8   GDPCAP   50418 non-null  float64
dtypes: float64(4), int32(2), object(3)
memory usage: 3.1+ MB


After viewing both datasets, it is evident that we have to create a new dataset by combining both the dataset. Before doing that let's have quick look at how many samples of each country are present in the dataset.

In [227]:
data['COUNTRY'].value_counts()

Afghanistan        294
Indonesia          294
Macedonia          294
Luxembourg         294
Lithuania          294
                  ... 
Tajikistan         172
Comoros            171
Lesotho            158
Hong Kong           51
Solomon Islands      4
Name: COUNTRY, Length: 210, dtype: int64

Now lets retrieve the highest value amongst these.

In [229]:
data2.shape

(50418, 14)

In [230]:
data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50418 entries, 0 to 50417
Data columns (total 14 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   iso_code                 50418 non-null  object 
 1   location                 50418 non-null  object 
 2   date                     50418 non-null  object 
 3   total_cases              47324 non-null  float64
 4   total_deaths             39228 non-null  float64
 5   stringency_index         43292 non-null  float64
 6   population               50418 non-null  int64  
 7   gdp_per_capita           44706 non-null  float64
 8   human_development_index  44216 non-null  float64
 9   Unnamed: 9               50418 non-null  object 
 10  Unnamed: 10              50418 non-null  object 
 11  Unnamed: 11              50418 non-null  object 
 12  Unnamed: 12              50418 non-null  float64
 13  Unnamed: 13              50418 non-null  object 
dtypes: float64(6), int64(1

In [231]:
data2.isnull().sum().sort_values(ascending = False)

total_deaths               11190
stringency_index            7126
human_development_index     6202
gdp_per_capita              5712
total_cases                 3094
iso_code                       0
location                       0
date                           0
population                     0
Unnamed: 9                     0
Unnamed: 10                    0
Unnamed: 11                    0
Unnamed: 12                    0
Unnamed: 13                    0
dtype: int64

In [232]:
data2['total_cases'].mean()

66219.27470205392

In [233]:
data2['total_cases'].fillna(method = 'ffill', inplace = True)
data2['total_cases'] = data2['total_cases'].astype(int)
data2['total_deaths'].fillna(method = 'ffill', inplace = True)
data2['total_deaths'] = data2['total_deaths'].astype(int)

In [234]:
data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50418 entries, 0 to 50417
Data columns (total 14 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   iso_code                 50418 non-null  object 
 1   location                 50418 non-null  object 
 2   date                     50418 non-null  object 
 3   total_cases              50418 non-null  int32  
 4   total_deaths             50418 non-null  int32  
 5   stringency_index         43292 non-null  float64
 6   population               50418 non-null  int64  
 7   gdp_per_capita           44706 non-null  float64
 8   human_development_index  44216 non-null  float64
 9   Unnamed: 9               50418 non-null  object 
 10  Unnamed: 10              50418 non-null  object 
 11  Unnamed: 11              50418 non-null  object 
 12  Unnamed: 12              50418 non-null  float64
 13  Unnamed: 13              50418 non-null  object 
dtypes: float64(4), int32(2

In [235]:
data2['stringency_index'].fillna(value = data2['stringency_index'].mean(), inplace = True) 
data2['gdp_per_capita'].fillna(method = 'ffill', inplace = True)
data2['human_development_index'].fillna(value = data2['human_development_index'].mean(), inplace = True)

In [236]:
data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50418 entries, 0 to 50417
Data columns (total 14 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   iso_code                 50418 non-null  object 
 1   location                 50418 non-null  object 
 2   date                     50418 non-null  object 
 3   total_cases              50418 non-null  int32  
 4   total_deaths             50418 non-null  int32  
 5   stringency_index         50418 non-null  float64
 6   population               50418 non-null  int64  
 7   gdp_per_capita           50418 non-null  float64
 8   human_development_index  50418 non-null  float64
 9   Unnamed: 9               50418 non-null  object 
 10  Unnamed: 10              50418 non-null  object 
 11  Unnamed: 11              50418 non-null  object 
 12  Unnamed: 12              50418 non-null  float64
 13  Unnamed: 13              50418 non-null  object 
dtypes: float64(4), int32(2

In [237]:
data2.head(10)

iso_code     location        date  total_cases  total_deaths  \
0      AFG  Afghanistan  2019-12-31            0             0   
1      AFG  Afghanistan  2020-01-01            0             0   
2      AFG  Afghanistan  2020-01-02            0             0   
3      AFG  Afghanistan  2020-01-03            0             0   
4      AFG  Afghanistan  2020-01-04            0             0   
5      AFG  Afghanistan  2020-01-05            0             0   
6      AFG  Afghanistan  2020-01-06            0             0   
7      AFG  Afghanistan  2020-01-07            0             0   
8      AFG  Afghanistan  2020-01-08            0             0   
9      AFG  Afghanistan  2020-01-09            0             0   

   stringency_index  population  gdp_per_capita  human_development_index  \
0               0.0    38928341        1803.987                    0.498   
1               0.0    38928341        1803.987                    0.498   
2               0.0    38928341        1803.987                    0.498   
3               0.0    38928341        1803.987                    0.498   
4               0.0    38928341        1803.987                    0.498   
5               0.0    38928341        1803.987                    0.498   
6               0.0    38928341        1803.987                    0.498   
7               0.0    38928341        1803.987                    0.498   
8               0.0    38928341        1803.987                    0.498   
9               0.0    38928341        1803.987                    0.498   

  Unnamed: 9 Unnamed: 10 Unnamed: 11  Unnamed: 12  Unnamed: 13  
0      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
1      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
2      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
3      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
4      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
5      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
6      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
7      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
8      #NUM!       #NUM!       #NUM!    17.477233  7.497754494  
9      #NUM!       #NUM!       #NUM!    17.477233  7.497754494

In [238]:
data2.columns

Index(['iso_code', 'location', 'date', 'total_cases', 'total_deaths',
       'stringency_index', 'population', 'gdp_per_capita',
       'human_development_index', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11',
       'Unnamed: 12', 'Unnamed: 13'],
      dtype='object')

In [239]:
data2.drop(['Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13'], axis = 1, inplace = True)

In [240]:
data2.head()

iso_code     location        date  total_cases  total_deaths  \
0      AFG  Afghanistan  2019-12-31            0             0   
1      AFG  Afghanistan  2020-01-01            0             0   
2      AFG  Afghanistan  2020-01-02            0             0   
3      AFG  Afghanistan  2020-01-03            0             0   
4      AFG  Afghanistan  2020-01-04            0             0   

   stringency_index  population  gdp_per_capita  human_development_index  
0               0.0    38928341        1803.987                    0.498  
1               0.0    38928341        1803.987                    0.498  
2               0.0    38928341        1803.987                    0.498  
3               0.0    38928341        1803.987                    0.498  
4               0.0    38928341        1803.987                    0.498

In [243]:
data['COUNTRY'].value_counts().mode()

0    294
Name: COUNTRY, dtype: int64

We will use the mode value for dividing the sum of all the samples of human development index, stringency index, population and the gdp. Let's create a new dataset by combining the necessary columns from both the datasets.


## Aggregating the data

In [244]:
code = data['CODE'].unique().tolist()
country = data['COUNTRY'].unique().tolist()
hdi = []
tc = []
td = []
sti = []
population = data['POP'].unique().tolist()
gdp = []

In [300]:
for i in country:
    hdi.append(data[data['COUNTRY'] == i]['HDI'].sum()/294)
    tc.append(data2[data2['location'] == i]['total_cases'].sum())
    td.append(data2[data2['location'] == i]['total_deaths'].sum())
    sti.append(data[data['COUNTRY'] == i]['STI'].sum()/294)
    population.append(data2[data2['location'] == i]['population'].sum()/294)
    gdp.append(data2[data2['location'] == i]['gdp_per_capita'].sum()/294)
    
aggregated_data = pd.DataFrame(list(zip(code, country, hdi, tc, td, sti, population, gdp)),
                               columns = ['Country Code', 'Country', 'HDI', 'Total Cases', 'Total Deaths',
                                          'Stringency Index', 'Population', 'Gdp During Covid'])
aggregated_data.head(10)

Country Code              Country       HDI  Total Cases  Total Deaths  \
0          AFG          Afghanistan  0.498000      5126433        165875   
1          ALB              Albania  0.600765      1071951         35532   
2          DZA              Algeria  0.754000      5866211        239352   
3          AND              Andorra  0.659551       223583         45114   
4          AGO               Angola  0.418952       304005         12292   
5          AIA             Anguilla  0.409071          614         49887   
6          ATG  Antigua and Barbuda  0.567755        12619          5629   
7          ARG            Argentina  0.707143     47157884       1077506   
8          ARM              Armenia  0.755000     65844325       2387268   
9          ABW                Aruba  0.554694       215174         32973   

   Stringency Index  Population  Gdp During Covid  
0          3.049673   17.477233       1803.987000  
1          3.005624   14.872537       9033.238010  
2          3.195168   17.596309      13913.839000  
3          2.677654   11.254996      10695.672156  
4          2.965560   17.307957       4196.370544  
5          2.326181    9.615939       4097.399541  
6          0.000000   11.491988      15643.067354  
7          3.475438   17.626514      16229.063143  
8          0.000000   14.901792       8787.580000  
9          2.768908   11.578395      26429.716653

Here I have added an extra column **Gdp Before Covid manually**. Since the dataset contains the values of **GDP During Covid** where the GDP values had dropped by 3.4% globally in the year 2020.

In [302]:
aggregated_data['Gdp Before Covid'] = aggregated_data['Gdp During Covid'] + aggregated_data['Gdp During Covid'] * 0.034

In [303]:
aggregated_data

Country Code      Country       HDI  Total Cases  Total Deaths  \
0            AFG  Afghanistan  0.498000      5126433        165875   
1            ALB      Albania  0.600765      1071951         35532   
2            DZA      Algeria  0.754000      5866211        239352   
3            AND      Andorra  0.659551       223583         45114   
4            AGO       Angola  0.418952       304005         12292   
..           ...          ...       ...          ...           ...   
205          VEN    Venezuela  0.566867      4842984         47215   
206          VNM      Vietnam  0.694000      2201946        159822   
207          YEM        Yemen  0.296721       228925         65039   
208          ZMB       Zambia  0.430000      1129913         35430   
209          ZWE     Zimbabwe  0.387602       593699         17022   

     Stringency Index  Population  Gdp During Covid  Gdp Before Covid  
0            3.049673   17.477233       1803.987000       1865.322558  
1            3.005624   14.872537       9033.238010       9340.368103  
2            3.195168   17.596309      13913.839000      14386.909526  
3            2.677654   11.254996      10695.672156      11059.325010  
4            2.965560   17.307957       4196.370544       4339.047143  
..                ...         ...               ...               ...  
205          3.235752   17.163165      12473.332714      12897.426027  
206          3.710868   18.393706       6171.884000       6381.728056  
207          2.369568   17.210890        971.004663       1004.018822  
208          2.645104   16.726989       2697.921650       2789.650986  
209          3.161132   16.514381       1376.367602       1423.164101  

[210 rows x 9 columns]

Here I have sorted out top 20 countries with the highest Covid Cases.

In [304]:
info = aggregated_data.sort_values(by = ['Total Cases'], ascending = False).head(20)
info

Country Code         Country       HDI  Total Cases  Total Deaths  \
200          USA   United States  0.924000    761184687      29139980   
91           IDN       Indonesia  0.694000    489687127       9168032   
27           BRA          Brazil  0.759000    426003311      14342127   
90           IND           India  0.640000    407893299       7248130   
157          RUS          Russia  0.816000    138661367       2642435   
150          PER            Peru  0.599490     75320488       3044986   
125          MEX          Mexico  0.774000     74353246       7296660   
178          ESP           Spain  0.887969     73808620       5514199   
92           IRN            Iran  0.798000     70877101       3552131   
8            ARM         Armenia  0.755000     65844325       2387268   
199          GBR  United Kingdom  0.922000     63174296       7280594   
175          ZAF    South Africa  0.608653     63135859       1362533   
42           COL        Colombia  0.581847     60725640       2016697   
40           CHL           Chile  0.656622     51268034       1285554   
97           ITA           Italy  0.880000     50752853       6664225   
68           FRA          France  0.901000     50419935       5649590   
151          PHL     Philippines  0.699000     48470198       1562716   
7            ARG       Argentina  0.707143     47157884       1077506   
145          PAK        Pakistan  0.562000     44115377        877637   
73           DEU         Germany  0.936000     42937034       1650211   

     Stringency Index  Population  Gdp During Covid  Gdp Before Covid  
200          3.350949   19.617637      54225.446000      56069.111164  
91           3.640149   19.426899      11188.744000      11569.161296  
27           3.136028   19.174732      14103.452000      14582.969368  
90           3.610552   21.045353       6426.674000       6645.180916  
157          3.380088   18.798668      24765.954000      25607.996436  
150          3.430126   17.311165       9781.040510      10113.595888  
125          3.019289   18.674802      17336.469000      17925.908946  
178          3.393922   17.660427      34155.787347      35317.084117  
92           3.207064   18.246243      19082.620000      19731.429080  
8            0.000000   14.901792       8787.580000       9086.357720  
199          3.353883   18.033340      39753.244000      41104.854296  
175          3.364333   17.898266      10705.742367      11069.737608  
42           3.357923   17.745037      10324.433065      10675.463789  
40           2.989194   16.766047      17733.508412      18336.447698  
97           3.629838   17.917523      35220.084000      36417.566856  
68           3.385794   17.994097      38605.671000      39918.263814  
151          3.678102   18.512175       7599.188000       7857.560392  
7            3.475438   17.626514      16229.063143      16780.851290  
145          3.385802   19.213186       5034.708000       5205.888072  
73           3.284516   18.243752      45229.245000      46767.039330

# Data Visualization

## Analyzing the Spread of Covid-19

Let's start by analyzing the spread of covid-19 in all the countries with the highest number of covid-19 cases.
First let's have a look at the countries with the highest number of covid-19 cases.

In [305]:
figure = px.bar(info, y = 'Total Cases', x = 'Country', title = 'Countries with Highest Covid Cases' )
figure.show()

It is evident that USA tops the chart with highest number of Covid-19 cases followed by Indonesia, Brazil and India.

Now let's have a look at the total number of deaths among the countries with the highest number of covid-19 cases.

In [306]:
figure = px.bar(info, x = 'Country', y = 'Total Deaths', title = 'Countries with Highest Covid Death Rate')
figure.show()

From the chart it is evident that USA is having the highest death rate, however we can see a slight change in the plot.
Brazil is having the second highest death rate followed by Indonesia. Here the order of the plot is disoriented 
becasue we had sorted the values based on Total Cases and not Total Deaths. 

Let's rearrange the values based on Total Deaths in order to have a proper view of the plot.

In [307]:
cov_info = aggregated_data.sort_values(by = ['Total Deaths'], ascending = False).head(20)
cov_info

Country Code                       Country       HDI  Total Cases  \
200          USA                 United States  0.924000    761184687   
27           BRA                        Brazil  0.759000    426003311   
91           IDN                     Indonesia  0.694000    489687127   
125          MEX                        Mexico  0.774000     74353246   
199          GBR                United Kingdom  0.922000     63174296   
90           IND                         India  0.640000    407893299   
97           ITA                         Italy  0.880000     50752853   
69           PYF              French Polynesia  0.658895        95388   
68           FRA                        France  0.901000     50419935   
178          ESP                         Spain  0.887969     73808620   
28           VGB        British Virgin Islands  0.534398         4605   
92           IRN                          Iran  0.798000     70877101   
150          PER                          Peru  0.599490     75320488   
179          LKA                     Sri Lanka  0.770000     26670332   
201          VIR  United States Virgin Islands  0.660000        98566   
157          RUS                        Russia  0.816000    138661367   
8            ARM                       Armenia  0.755000     65844325   
42           COL                      Colombia  0.581847     60725640   
158          RWA                        Rwanda  0.390327       388775   
18           BEL                       Belgium  0.909769     18261898   

     Total Deaths  Stringency Index  Population  Gdp During Covid  \
200      29139980          3.350949   19.617637      54225.446000   
27       14342127          3.136028   19.174732      14103.452000   
91        9168032          3.640149   19.426899      11188.744000   
125       7296660          3.019289   18.674802      17336.469000   
199       7280594          3.353883   18.033340      39753.244000   
90        7248130          3.610552   21.045353       6426.674000   
97        6664225          3.629838   17.917523      35220.084000   
69        5925689          0.000000   12.545768      28232.038316   
68        5649590          3.385794   17.994097      38605.671000   
178       5514199          3.393922   17.660427      34155.787347   
28        3693903          2.240659   10.316822       9929.981510   
92        3552131          3.207064   18.246243      19082.620000   
150       3044986          3.430126   17.311165       9781.040510   
179       3008083          3.212534   16.879520      11669.077000   
201       2857595          0.000000   11.556205      38732.461429   
157       2642435          3.380088   18.798668      24765.954000   
8         2387268          0.000000   14.901792       8787.580000   
42        2016697          3.357923   17.745037      10324.433065   
158       1864143          3.092047   16.376777       1381.197990   
18        1863491          3.420055   16.265620      42368.381605   

     Gdp Before Covid  
200      56069.111164  
27       14582.969368  
91       11569.161296  
125      17925.908946  
199      41104.854296  
90        6645.180916  
97       36417.566856  
69       29191.927619  
68       39918.263814  
178      35317.084117  
28       10267.600882  
92       19731.429080  
150      10113.595888  
179      12065.825618  
201      40049.365117  
157      25607.996436  
8         9086.357720  
42       10675.463789  
158       1428.158721  
18       43808.906580

In [308]:
figure = px.bar(cov_info, x = 'Country', y = 'Total Deaths', title = 'Countries with Highest Covid Death rate')
figure.show()

As we have seen the total number of cases and deaths. Let's comapre both in all these top 20 countries.

In [309]:
fig = go.Figure()
fig.add_trace(go.Bar(
     x = info['Country'], 
     y = info['Total Cases'],
     name = 'Total Cases',
     marker_color = 'indianred'
))
fig.add_trace(go.Bar(
    x = info['Country'],
    y = info['Total Deaths'],
    name = 'Total Deaths',
    marker_color = 'lightsalmon'
))
fig.update_layout(barmode = 'group', xaxis_tickangle = -45)
fig.show()

The above plot shows the comparison of total cases and total deaths in each country. 

Now let's have a look at the percentage of total deaths an dtotal cases among all the countries with the highest number of covid-19 cases:

In [310]:
cases = info['Total Cases'].sum()
deceased = info['Total Deaths'].sum()

labels = ['Total Cases', 'Total Deaths']
values = [cases, deceased]

fig = px.pie(info, values = values, names = labels, title = 'Percentage of Total Cases and Deaths', hole = 0.7)
fig.show()

Calculating the death rate for each top 20 country

In [311]:
for i in info['Country']:
    death_rate = info[info['Country'] == i]['Total Deaths'].sum()/ info[info['Country'] == i]['Total Cases'].sum()
    print(f"Death rate of {i} = ", death_rate)
    


Death rate of United States =  0.038282404385783446
Death rate of Indonesia =  0.018722223833341652
Death rate of Brazil =  0.03366670312099992
Death rate of India =  0.01776967167092392
Death rate of Russia =  0.01905674995977791
Death rate of Peru =  0.04042706149221975
Death rate of Mexico =  0.09813505653808309
Death rate of Spain =  0.07470941740951124
Death rate of Iran =  0.05011676479262322
Death rate of Armenia =  0.03625624531802855
Death rate of United Kingdom =  0.11524614377974232
Death rate of South Africa =  0.02158096874867894
Death rate of Colombia =  0.03320997522628004
Death rate of Chile =  0.02507515696818021
Death rate of Italy =  0.13130739665019422
Death rate of France =  0.11205071962111811
Death rate of Philippines =  0.032240759569416244
Death rate of Argentina =  0.022848904755777423
Death rate of Pakistan =  0.019894128979108576
Death rate of Germany =  0.03843327883337261


Another important column in this dataset is the stringency index. It is a composite measure of response indicatrs, including school closures, workplace closures and travel bans. It shows how strictly countries are following these measures to control the spread of covid-19:

In [313]:
fig = px.bar(info, x = 'Country', y = 'Total Cases', 
             hover_data = ['Population', 'Total Deaths'],
             color = 'Stringency Index', height = 400,
             title = 'Stringency Index during Covid-19')
fig.show()

From the plot it is evident that amongst the top 20 countries,
1. Indonesia is performing well in the stringency index during the outbreak with an overall stringency index value of 3.64
2. India follow Indonesia in the stringency index with an overall value of 3.61
3. Russia has a stringency index of 3.38
4. Armenia has a stringency index of 0 thus stating that the spread the virus is more when compared to other countries.

# Analyzing the impact on Economy

Here the GDP per capita is the primary factor for analyzing the economic slowdowns caused due to the outbreak of covid-19. 
Let's have a look at the GDP per capita before the outbreak among the countries with the highest number of covid-19 cases:

In [315]:
fig = px.bar(info, x = 'Country', y = 'Total Cases', 
             hover_data = ['Population', 'Total Deaths'],
             color = 'Gdp Before Covid', height = 400,
             title = 'GDP Per Capita Before Covid-19')
fig.show()

Now let's have a look at the GDP per capita during the rise in the case of covid-19:


In [316]:
fig = px.bar(info, x = 'Country', y = 'Total Cases', 
             hover_data = ['Population', 'Total Deaths'],
             color = 'Gdp During Covid', height = 400,
             title = 'GDP Per Capita During Covid-19')
fig.show()

Now let's compare the GDP per capita before covid-19 and during covid-19 to have a look at the impact of covid-19 on the GDP per capita:

In [318]:
fig = go.Figure()
fig.add_trace(go.Bar(
    x = info['Country'],
    y = info['Gdp Before Covid'],
    name = 'GDP per capita before covid-19',
    marker_color = 'indianred'
))
fig.add_trace(go.Bar(
    x = info['Country'],
    y = info['Gdp During Covid'],
    name = 'GDP per capita during covid-19',
    marker_color = 'lightsalmon'
))
fig.update_layout(barmode = 'group', xaxis_tickangle = -45)
fig.show()

We can see drop in GDP per capita in all the countries with the highest number of covid-19 cases.

One other important factor is the Human Development Index. It is a statistic composite index of life expectency, education and per capita indicators. Let's have a look at how many countries were spending their budget on the human development:


In [319]:
fig = px.bar(info, x = 'Country', y='Total Cases', hover_data = ['Population', 'Total Deaths'],
             color = 'HDI', height = 400, title = 'Human Development Index during Covid-19')
fig.show()

# Conslusion

We studied the spread of covid-19 among the countries and its impact on the global economy. We saw that the outbreak of covid-19 resulted in highest number of covid-19 cases and deaths in the United States. One major reason behind this the stringency index of the United States. It is comparitively low when compared with the population. we also analyzed how the GDP per capita of every top 20 countries was affected during the outbreak of covid-19.